core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -191,7 +191,7 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
    */
   def registerKryoClasses(classes: Array[Class[_]]): SparkConf = {
     val allClassNames = new LinkedHashSet[String]()
-    allClassNames ++= get("spark.kryo.classesToRegister", "").split(',').filter(!_.isEmpty)
+    allClassNames ++= get("spark.kryo.classesToRegister", "").split(',').trim.filter(!_.isEmpty)
```

Comment:
Isn't `trim` being applied to `Array[String]`?

I think this should be `.split(',').map(_.trim)`

```scala
scala> "A, B,   C".split(",").trim
<console>:8: error: value trim is not a member of Array[String]
              "A, B,   C".split(",").trim
                                     ^
```

```scala
scala> "A, B,   C".split(",").map(_.trim)
res2: Array[String] = Array(A, B, C)
```